namenode/namenode.go

```diff
@@ -13,17 +14,24 @@ type iNode struct {
 	rwlock         *sync.RWMutex // Per-file rwlock
 }
 
+type DataNodeInfo struct {
+	heartbeatReceived bool // Whether received heartbeat during the last interval
+	isAlive           bool
+}
+
 type NameNode struct {
-	inodes       map[string]iNode
-	globalRWLock *sync.RWMutex // The RWLock for the entire namenode
+	inodes         map[string]iNode
+	inodeRWLock    *sync.RWMutex // The RWLock for the inodes
+	datanodeRWLock *sync.RWMutex // The RWLock for the datanodes
+	datanodes      map[string]DataNodeInfo
```

Comment:
This is a minor issue: for the sake of visual consistency, I suggest putting datanodes before datanodeRWLock

Reply:
Actually, you can also add an empty line between the first two fields and the other, which separates them into groups more clearly